types/react-blessed/index.d.ts

```diff
@@ -246,8 +240,50 @@ export type BlessedIntrinsicElementsPrefixed = {
 // elements will collide with ones set in react defs.
 
 // augment react JSX when old JSX transform is used
-declare module "react" {
+declare global {
```

Comment:
Could you elaborate on this change? I wonder about the exact priority for JSX namespaces, like - what's being picked first etc. 

Without rechecking this stuff, I can't quickly figure out how this was previously working and how it's working now. One thought train is this:
1. previously they declared a JSX namespace within `"react"` module and that was picked first, essentially overriding the global JSX namespace
2. but then... now this declares a global JSX namespace and that merges with the global one declared in React? I'm not sure how to reason about such a merge

Reply:
> essentially overriding the global JSX namespace

Yep. Replacing would be another word. Just to emphasize this: By declaring a scoped JSX namespace you would disable all declarations relevant to JSX that were done in `react`. So in that sense you could potentially prevent us from ever shipping a scoped namespace. But I think generally adding stuff via module augmentation should not be considered supported. Otherwise every interface or type we add to the module would be considered a breaking change.

> now this declares a global JSX namespace and that merges with the global one declared in React? I'm not sure how to reason about such a merge

The global JSX namespace currently flows into the scoped one i.e. the interfaces in the scoped namespace just extend their global counterparts. We need to do that so that current augmentation still works. If we wouldn't do that, all existing module augmentation (`declare global`) would stop functioning.

I think this diff is still outdated since you should be able to augment both `module 'react'` or `global` and get TS to recognize JSX augmentations. I'll double check.

Reply:
Confirmed that this was only needed when we didn't have backwards compat for global module augmentation. Now we do.

To clarify: The global one and augmenting it is deprecated. It should continue to work just like before. In v19 we'll drop it and `react-blessed` will already worked since they always augmented only the scoped one.